src/app/components/layouts/Auth.tsx

```diff
@@ -14,11 +17,20 @@ const Auth = ({isSignIn}: Props) => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
 
+  const { loginUser } = UserAuth();
+  const router = useRouter();
+
   // handling submit email and password
-  const handleSubmit = (e:React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (e:React.FormEvent<HTMLFormElement>) => {
```

Comment:
I am thinking about that how about saving and using the event type in different files with using interfaces or types? Perhaps in typescript we often declare event types or something like that, so this will save us from having to modify individual types when refactoring in the future.

this down below is an example.
```
type Props = {
  onClick: (event: React.MouseEvent<HTMLInputElement>) => void
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  onkeypress: (event: React.KeyboardEvent<HTMLInputElement>) => void
  onBlur: (event: React.FocusEvent<HTMLInputElement>) => void
  onFocus: (event: React.FocusEvent<HTMLInputElement>) => void
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void
  onClickDiv: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
}
```